vortex-array/src/builders/primitive.rs

```diff
@@ -95,6 +95,12 @@ impl<T: NativePType> PrimitiveBuilder<T> {
     }
 
     pub fn finish_into_primitive(&mut self) -> VortexResult<PrimitiveArray> {
+        assert_eq!(
+            self.nulls.len(),
+            self.values.len(),
+            "null count must equal value count"
+        );
```

Comment:
Can you return an error here?

Reply:
not quite sure why this returns result otherwise

Reply:
Hmm. This is programmer error right? So shouldn't it be an assertion? I can remove the `VortexResult` though.

Reply:
Done in this PR: https://github.com/spiraldb/vortex/pull/2392